styles/layers.h

```diff
@@ -4,21 +4,31 @@
 #include "alpha.h"
 #include "../functions/int.h"
 #include "colors.h"
+#include "../common/color.h"
+#include <utility>
 
 // Usage: Layers<BASE, LAYER1, LAYER2, ...>
 // BASE: COLOR or LAYER
 // LAYER1, LAYER2: LAYER
 // return value: COLOR or LAYER (same as BASE)
-// This style works like layers in gimp or photoshop.
-// In most cases, the layers are expected to be normally transparent effects
-// that turn opaque when then want to paint an effect over the base color.
-// If the base color is opqaque, the final result of this style will also be
-// opaque. If the base color is transparent, the final result may also be transparent,
-// depending on what the layers paint on top of the base color.
+// This style works like layers in Photoshop: each layer is painted over the one beneath it.
+// When a Layers<> is used as the first template argument to StylePtr<>, its BASE color (the first layer)
+// must be a solid (opaque) color, not transparent. No additional solid colors may be stacked on top
+// of that solid base.
+// Additional layers stacked on top of a solid base color are transparent,
+// so the base color shows through until an effect wants to paint over it (e.g., a clash).
+// Layers<> can be nested inside other Layers<> and can be used anywhere a STYLE or COLOR is expected.
+
 
 template<class BASE, class L1>
 class Compose {
 public:
+  using _BaseColorT  = decltype(std::declval<BASE&>().getColor(0));
+  using _LayerColorT = decltype(std::declval<L1&>().getColor(0));
+  static_assert(!(color_details::IsOpaqueColor<_BaseColorT>::value &&
```

Comment:
I think it's enough to just check if _LayerColorT is opaque.
It doesn't matter if _BaseColorT is opaque or not.